config/management/src/smoke_test.rs

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{layout::Layout, storage_helper::StorageHelper};
+use config_builder::{BuildSwarm, SwarmConfig};
+use executor::db_bootstrapper;
+use libra_config::config::{
+    DiscoveryMethod, NetworkConfig, NetworkKeyPairs, NodeConfig, OnDiskStorageConfig, RoleType,
+    SecureBackend,
+};
+use libra_crypto::{ed25519::Ed25519PrivateKey, x25519, Uniform};
+use libra_secure_storage::Value;
+use libra_swarm::swarm::{LibraSwarm, LibraSwarmDir};
+use libra_temppath::TempPath;
+use libra_types::account_address::AccountAddress;
+use libra_vm::LibraVM;
+use libradb::LibraDB;
+use std::{convert::TryInto, path::PathBuf};
+use storage_interface::DbReaderWriter;
+
+struct ManagementBuilder {
+    configs: Vec<NodeConfig>,
+    faucet_key: Ed25519PrivateKey,
+}
+
+impl BuildSwarm for ManagementBuilder {
+    fn build_swarm(&self) -> anyhow::Result<(Vec<NodeConfig>, Ed25519PrivateKey)> {
+        Ok((self.configs.clone(), self.faucet_key.clone()))
+    }
+}
+
+#[test]
+fn smoke_test() {
+    let helper = StorageHelper::new();
+    let num_validators = 5;
+    let shared = "_shared";
+    let association = "association";
+    let association_shared = association.to_string() + shared;
+
+    // Step 1) Prepare the layout
+    let mut layout = Layout::default();
+    layout.association = vec![association_shared.to_string()];
+    layout.operators = (0..num_validators)
+        .map(|v| (v.to_string() + shared))
+        .collect();
+
+    let mut common_storage = helper.storage(crate::constants::COMMON_NS.into());
+    let layout_value = Value::String(layout.to_toml().unwrap());
+    common_storage
+        .set(crate::constants::LAYOUT, layout_value)
+        .unwrap();
+
+    // Step 2) Set association key
+    helper.initialize(association.into());
+    helper
+        .association_key(&association, &association_shared)
+        .unwrap();
+
+    // Step 3) Prepare validators
+    let mut configs = Vec::new();
+    for i in 0..num_validators {
+        let storage = helper.storage(i.to_string());
+        let ns = i.to_string();
+        let ns_shared = ns.clone() + shared;
+        helper.initialize(ns.clone());
+
+        let mut config = NodeConfig::default();
+
+        let mut network = NetworkConfig::default();
+        network.discovery_method = DiscoveryMethod::Onchain;
+        config.validator_network = Some(network);
+
+        let mut network = NetworkConfig::default();
+        network.discovery_method = DiscoveryMethod::Onchain;
+        config.full_node_networks = vec![NetworkConfig::default()];
```

Comment:
woops, good call, I added fullnode support but don't use it, so it is untested... I'll clean this up